datafusion/physical-plan/src/windows/mod.rs

```diff
@@ -257,22 +266,17 @@ fn create_built_in_window_expr(
                 return exec_err!("NTILE requires a positive integer, but finds NULL");
             }
 
-            if n.is_unsigned() {
-                let n: u64 = n.try_into()?;
-                Arc::new(Ntile::new(name, n, out_data_type))
-            } else {
-                let n: i64 = n.try_into()?;
-                if n <= 0 {
-                    return exec_err!("NTILE requires a positive integer");
-                }
-                Arc::new(Ntile::new(name, n as u64, out_data_type))
+            let n: i64 = get_integer(n)?;
+            if n <= 0 {
+                return exec_err!("NTILE requires a positive integer");
             }
+            Arc::new(Ntile::new(name, n as u64, out_data_type))
         }
         BuiltInWindowFunction::Lag => {
             let arg = args[0].clone();
             let shift_offset = get_scalar_value_from_args(args, 1)?
-                .map(|v| v.try_into())
-                .and_then(|v| v.ok());
+                .map(get_integer)
+                .map_or(Ok(None), |v| v.map(Some))?;
```

Comment:
this causes us to error out if the arg is given but cannot be converted into an i64, instead of silently returning None and then defaulting to 1